src/mds/Capability.h

```diff
@@ -381,7 +381,7 @@ class Capability : public Counter<Capability> {
   ceph_seq_t mseq = 0;
 
   int suppress = 0;
-  unsigned state = 0;
+  uint32_t state = 0;
```

Comment:
ILP32 and LP64 architectures all implement `unsigned` as `unsigned int` a.k.a. `uint32_t`
could you point where this standard isn't being followed ?

however, I agree its always a good idea to specify the data size upfront during serialization/deserialization

Reply:
I couldn't tell exactly where. I just found this when I was coding the kclient patches and googled about this. It's said in some platform or OS it maybe different. As @gregsfortytwo mentioned we should always use explicitly sized type in these cases.

Reply:
so is `uint32_t` good enough or is `__le32` more appropriate ?

Reply:
The `uint32_t` is good enough here and the `encode()` will help switch it to `__le32`.